go/libraries/doltcore/table/typed/parquet/reader.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package parquet
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/xitongsys/parquet-go-source/local"
+	"github.com/xitongsys/parquet-go/common"
+	"github.com/xitongsys/parquet-go/reader"
+	"github.com/xitongsys/parquet-go/source"
+
+	"github.com/dolthub/dolt/go/libraries/doltcore/row"
+	"github.com/dolthub/dolt/go/libraries/doltcore/schema"
+	"github.com/dolthub/dolt/go/store/types"
+)
+
+// ParquetReader implements TableReader.  It reads parquet files and returns rows.
+type ParquetReader struct {
+	fileReader     source.ParquetFile
+	pReader        *reader.ParquetReader
+	sch            schema.Schema
+	vrw            types.ValueReadWriter
+	numRow         int
+	rowReadCounter int
+	fileData       map[string][]interface{}
+	columnName     []string
+}
+
+// OpenParquetReader opens a reader at a given path within local filesystem.
+func OpenParquetReader(vrw types.ValueReadWriter, path string, sch schema.Schema) (*ParquetReader, error) {
+	fr, err := local.NewLocalFileReader(path)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewParquetReader(vrw, fr, sch)
+}
+
+// NewParquetReader creates a ParquetReader from a given fileReader.
+// The ParquetFileInfo should describe the parquet file being read.
+func NewParquetReader(vrw types.ValueReadWriter, fr source.ParquetFile, sche schema.Schema) (*ParquetReader, error) {
+	pr, err := reader.NewParquetColumnReader(fr, 4)
+	if err != nil {
+		return nil, err
+	}
+
+	columns := sche.GetAllCols().GetColumns()
+	num := pr.GetNumRows()
+
+	data := make(map[string][]interface{})
```

Comment:
Having all the column in memory like this could be a problem in the future. Please leave a TODO